corehq/apps/app_manager/models.py

```diff
@@ -3899,6 +3914,7 @@ def validate_app(self):
                 if xmlns_count[xmlns] > 1:
                     errors.append({'type': "duplicate xmlns", "xmlns": xmlns})
 
+        self.ensure_module_unique_ids(should_save=True)
```

Comment:
I'm still not very comfortable with this since we're introducing a side effect in a method that shouldn't really have one. I think I would be more happy with raising an exception that gets caught in the view and handled there:
```python
try:
    errors = app.validate_app()
except MissingModuleIdException:
    # comment about why this might happen
    app.ensure_module_unique_ids(should_save=True)
    errors = app.validate_app()
```